src/project/internal/notationproject.cpp

```diff
@@ -779,6 +795,43 @@ Ret NotationProject::saveSelectionOnScore(const muse::io::path_t& path)
     return ret;
 }
 
+Ret NotationProject::checkSavedFileForCorruption(engraving::MscIoMode ioMode, QString path, QString scoreFileName)
+{
+    if (ioMode != MscIoMode::Zip) {
+        return muse::make_ok();
+    }
+
+    // TODO: Execute this on Windows only?
+
+    MscReader::Params params;
+    params.filePath = path;
+    params.mainFileName = scoreFileName;
+    params.mode = MscIoMode::Zip;
+
+    MscReader msczReader(params);
+    Ret ret = msczReader.open();
+    if (!ret) {
+        return Ret(static_cast<int>(Err::CorruptionUponSavingError),
+                  muse::mtrc("project/save", "File “%1” could not be opened. %2")
+                  .arg(path)
+                  .arg(String(ret.toString().c_str()))
+                  .toStdString());
+    }
+
+    // Try to extract the main score file.
+    ByteArray scoreFile = msczReader.readScoreFile();
+    msczReader.close();
+
+    if (scoreFile.empty()) {
```

Comment:
is empty enough? Surely we should do a basic check that it's well-formed xml? Even constructing an XmlReader on the data is quite fast for a very large score, though it would cause some perceptible delay (<500ms).

Reply:
Yes, it is debatable how much deep we want to go with the verification. From what I've read and am thinking, I believe it should be enough but let's hear other thoughts on this. Note: `scoreFile` will be empty not only when the score file is empty in the zip file but when it cannot be read for whatever reason. I didn't see has_error() in MscReader.